src/edit/items/BoundWithFieldRow.js

```diff
@@ -0,0 +1,53 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
+import { Loading } from '@folio/stripes/components';
+
+import FieldRow from '../../components/RepeatableField/FieldRow';
+import useHoldingsQueryByHrids from '../../hooks/useHoldingsQueryByHrids';
+import useInstancesQuery from '../../hooks/useInstancesQuery';
+
+// Fetches holdings and instances info given in the input holdings HRID
+const BoundWithFieldRow = ({ fields, ...rest }) => {
+  const data = fields.value ? fields.value : [];
+
+  // Load the holdings records matching the input HRIDs
+  const holdingsRecordHrids = data.map(boundWithTitle => boundWithTitle.briefHoldingsRecord.hrid);
+  const { isLoading: isHoldingsLoading, holdingsRecords } = useHoldingsQueryByHrids(holdingsRecordHrids);
+
+  // Load the instance records matching the holdings' instanceIds.
+  const instanceIds = holdingsRecords.map(record => record.instanceId);
+  const instances = useInstancesQuery(instanceIds);
+  if (isHoldingsLoading || !instances.isSuccess) return <Loading size="large" />;
+
+  // Enrich the data render in the FieldRow
+  data.map(boundWithTitle => {
+    if (!boundWithTitle.briefHoldingsRecord?.id) {
+      const holdingsRecord = holdingsRecords.find(
```

Comment:
@maccabeelevine I think you could avoid `find` here if the `keyBy` from lodash is used above (line 22 maybe):

`const holdingsRecordsByHrid = keyBy(holdingsRecords, 'hrid');`

Then here you could do:

`const holdingsRecord  = holdingsRecordsByHrid[boundWithTitle?.briefHoldingsRecord?.hrid]`

Reply:
Thanks, I have to get more familar with lodash -- this is much more efficient.